modules/fedimint-ln/src/db.rs

```diff
@@ -118,3 +118,143 @@ impl_db_lookup!(
     key = LightningGatewayKey,
     query_prefix = LightningGatewayKeyPrefix
 );
+
+#[cfg(test)]
+mod fedimint_migration_tests {
+    use std::collections::BTreeMap;
+    use std::env;
+    use std::path::Path;
+
+    use fedimint_core::core::LEGACY_HARDCODED_INSTANCE_ID_LN;
+    use fedimint_core::db::apply_migrations;
+    use fedimint_core::module::DynModuleGen;
+    use fedimint_testing::apply_to_databases;
+    use futures::StreamExt;
+    use strum::IntoEnumIterator;
+
+    use crate::db::{
+        AgreedDecryptionShareKeyPrefix, ContractKeyPrefix, ContractUpdateKeyPrefix, DbKeyPrefix,
+        LightningGatewayKeyPrefix, ProposeDecryptionShareKeyPrefix,
+    };
+    use crate::LightningGen;
+
+    #[tokio::test(flavor = "multi_thread")]
+    async fn verify_lightning_database_migration() {
+        // Only run the database migration test if the database backup directory
+        // environment variable has been defined
+        let backup_dir = env::var("FM_TEST_DB_BACKUP_DIR");
+        if let Ok(backup_dir) = backup_dir {
```

Comment:
The indentation level here is getting out of control while the whole block is very long, so maybe `let Some(backup_dir) = backup_dir else { return };` https://doc.rust-lang.org/beta/rust-by-example/flow_control/let_else.html could help.

Reply:
I think the indendation is much better after reducing the scope of the PR, hopefully its easier to read